connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/HttpCommonResultResponseHandler.java

```diff
@@ -88,7 +88,30 @@ public HttpCommonResult handleResponse(ClassicHttpResponse response) {
     return new HttpCommonResult(code, headers, null, reason);
   }
 
-  private Document handleFileResponse(Map<String, String> headers, byte[] content) {
+  private Map<String, Object> formatHeaders(Header[] headersArray) {
+    return Arrays.stream(headersArray)
+        .collect(
+            Collectors.toMap(
+                Header::getName,
+                header -> {
+                  if (isHeaderGroupingEnabled && header.getName().equalsIgnoreCase("Set-Cookie")) {
+                    return new ArrayList<String>(List.of(header.getValue()));
+                  }
+                  return header.getValue();
+                },
+                (existingValue, newValue) -> {
+                  if (isHeaderGroupingEnabled
+                      && existingValue instanceof List
+                      && ((List<?>) existingValue).getFirst() instanceof String
+                      && newValue instanceof List
+                      && ((List<?>) newValue).getFirst() instanceof String) {
+                    ((List<String>) existingValue).add(((List<String>) newValue).getFirst());
+                  }
+                  return existingValue;
```

Comment:
Again, matter of taste. An else would improve readability.

Reply:
I added it in the case above, but will not add it here, as here the if case does not have a return statement and i would need to copy/dulicate this if moving it to a else block